main.go

```diff
@@ -3,13 +3,38 @@ package main
 import (
 	"log"
 	"net/http"
+	"net/http/httputil"
+	"net/url"
 	"observability-remote-write-proxy/pkg/remotewrite"
 )
 
+const (
+	prefixHeader = "X-Forwarded-Prefix"
+)
+
 func main() {
-	handler := http.HandlerFunc(handleRequestSuccess)
-	http.Handle("/", handler)
-	err := http.ListenAndServe(":8080", nil)
+	upstreamUrl, err := url.Parse("http://localhost:9091/api/v1/write")
+	if err != nil {
+		panic(err)
+	}
+
+	proxy := httputil.ReverseProxy{
+		Director: func(request *http.Request) {
+			request.URL.Scheme = upstreamUrl.Scheme
+			request.Host = upstreamUrl.Host
+			request.URL.Host = upstreamUrl.Host
+			request.URL.Path = upstreamUrl.Path
+			request.Header.Add(prefixHeader, "/")
+		},
+		ErrorHandler: func(writer http.ResponseWriter, request *http.Request, err error) {
+			log.Println(err)
+		},
+	}
+
+	proxy.Transport = http.DefaultTransport
+	http.Handle("/", http.HandlerFunc(handleRequestSuccess))
```

Comment:
What does `handleRequestSuccess` do in case of success? 
Looking at https://github.com/bf2fc6cc711aee1a0c2a/observability-remote-write-proxy/blob/bdeb90088e1c682333793762afb1296a0d4ed62b/main.go#L66 it seems to write a successfully response. This makes me think that we won't proxy the request to remote prometheus as intended in https://github.com/bf2fc6cc711aee1a0c2a/observability-remote-write-proxy/blob/bdeb90088e1c682333793762afb1296a0d4ed62b/main.go#L36

Reply:
Looking at the `handleRequestSuccess` code: it looks to be performing some sort of validation.
Maybe we can name it `validateRequest`

Reply:
@machi1990 handleRequestSuccess() should be decoding the write request and validating the cluster IDs (in case different clusters are sending a request), validateRequest() sounds good. thanks!

Reply:
Thanks @valeriiashapoval 

I am thinking ghttps://github.com/bf2fc6cc711aee1a0c2a/observability-remote-write-proxy/blob/bdeb90088e1c682333793762afb1296a0d4ed62b/main.go#L66 might need to be gone